spotbugs/src/main/java/edu/umd/cs/findbugs/detect/ConstructorThrow.java

```diff
@@ -0,0 +1,151 @@
+     * Also we are checking for final declaration on the class, or a final finalizer, as if present
+     * no finalizer attack can happen.
+     */
+    @Override
+    public void visit(JavaClass obj) {
+        resetState();
+        if (obj.isFinal()) {
+            isFinalClass = true;
+            return;
+        }
+        for (Method m : obj.getMethods()) {
+            // First visit the constructor, it might not be at the start of the class.
+            if ("<init>".equals(m.getName())) {
+                // This will visit all constructors.
+                doVisitMethod(m);
+            } else if ("finalize".equals(m.getName())) {
+                // Check for finalizer.
+                if (m.isFinal()) {
+                    isFinalFinalizer = true;
+                }
+            }
+        }
+        isFirstPass = false;
+    }
+
+    @Override
+    public void visit(Method obj) {
+        if (isFinalClass || isFinalFinalizer) {
+            return;
+        }
+        if (isConstructor() || methodCalledFromCtor()) {
+            // Check if there is a throws keyword for checked exceptions.
+            ExceptionTable tbl = obj.getExceptionTable();
+            boolean throwsExceptions = tbl != null && tbl.getNumberOfExceptions() > 0;
+            if (throwsExceptions) {
+                accumulateBug();
+            }
+        }
+    }
+
+    @Override
+    public void visit(Code obj) {
+        super.visit(obj);
+        bugAccumulator.reportAccumulatedBugs();
+    }
+
+    /**
+     * 1. Check for any throw expression in the constructor.
+     * 2. Check for any unchecked exception throw inside constructor,
+     *    or any of the called methods.
+     * If the class is final, we are fine, no finalizer attack can happen.
+     * In the first pass the detector shouldn't report, because there could be
+     * a final finalizer and a throwing constructor. Reporting in this case
+     * would be a false positive as classes with a final finalizer are not
+     * vulnerable to the finalizer attack.
+     */
+    @Override
+    public void sawOpcode(int seen) {
+        if (isFinalClass || isFinalFinalizer) {
+            return;
+        }
+        if (isFirstPass) {
+            tryCollectMethod(seen);
+        } else {
+            if (isConstructor() || methodCalledFromCtor()) {
+                if (seen == Const.ATHROW)
+                    accumulateBug();
+            }
+        }
+    }
+
+    /** Collects all the methods that are called from the constructor */
+    private void tryCollectMethod(int seen) {
+        if (!isMethodCall(seen)) {
+            return;
+        }
+        String classConstantOperand = "";
+        try {
+            classConstantOperand = getClassConstantOperand();
+        } catch (IllegalStateException e) {
+            bugReporter.logError("Seen OPcode and failed to get ClassConstantOperand: " + seen, e);
```

Comment:
Better to `return` in the case of catching exception? Or `calledFromCtor.add(method)` will be called always.

Reply:
Added return statement after logging